src/mam4xx/mo_setsox.hpp

```diff
@@ -0,0 +1,1364 @@
+const int pver = mam4::nlev;
+const int nmodes = AeroConfig::num_modes();
+const int nspec_gas = AeroConfig::num_gas_phase_species();
+const int modeptr_accum = int(ModeIndex::Accumulation);
+
+// these are the entries/indices in the gas-phase species array
+// from: eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/mo_sim_dat.F90#L23)
+// solsym(: 31) = (/ 'O3', 'H2O2', 'H2SO4', 'SO2', 'DMS',  'SOAG', 'so4_a1',
+//                   'pom_a1', 'soa_a1', 'bc_a1',  'dst_a1', 'ncl_a1',
+//                   'mom_a1', 'num_a1', 'so4_a2',  'soa_a2', 'ncl_a2',
+//                   'mom_a2', 'm_a2', 'dst_a3',  'ncl_a3', 'so4_a3', 'bc_a3',
+//                   'pom_a3', 'soa_a3', 'mom_a3', 'num_a3', 'pom_a4',
+//                   'bc_a4', 'mom_a4', 'num_a4' /)
+// Fortran-indexed:
+// 1: 'O3', 2: 'H2O2', 3: 'H2SO4', 4: 'SO2', 5: 'DMS', 6: 'SOAG', 7: 'so4_a1',
+// 8: 'pom_a1', 9: 'soa_a1', 10: 'bc_a1', 11: 'dst_a1', 12: 'ncl_a1',
+// 13: 'mom_a1', 14: 'num_a1', 15: 'so4_a2', 16: 'soa_a2', 17: 'ncl_a2',
+// 18: 'mom_a2', 19: 'num_a2', 20: 'dst_a3', 21: 'ncl_a3', 22: 'so4_a3',
+// 23: 'bc_a3', 24: 'pom_a3', 25: 'soa_a3', 26: 'mom_a3', 27: 'num_a3',
+// 28: 'pom_a4', 29: 'bc_a4', 30: 'mom_a4', 31: 'num_a4'
+// c++ indexed:
+// 0: 'O3', 1: 'H2O2', 2: 'H2SO4', 3: 'SO2', 4: 'DMS', 5: 'SOAG', 6: 'so4_a1',
+// 7: 'pom_a1', 8: 'soa_a1', 9: 'bc_a1', 10: 'dst_a1', 11: 'ncl_a1',
+// 12: 'mom_a1', 13: 'num_a1', 14: 'so4_a2', 15: 'soa_a2', 16: 'ncl_a2',
+// 17: 'mom_a2', 18: 'num_a2', 19: 'dst_a3', 20: 'ncl_a3', 21: 'so4_a3',
+// 22: 'bc_a3', 23: 'pom_a3', 24: 'soa_a3', 25: 'mom_a3', 26: 'num_a3',
+// 27: 'pom_a4', 28: 'bc_a4', 29: 'mom_a4', 30: 'num_a4'
+constexpr int id_so2 = 3;
+constexpr int id_h2o2 = 1;
+constexpr int id_o3 = 0;
+constexpr int id_h2so4 = 2;
+
+// ===========================================================================
+//     BAD CONSTANTS (let's just put the global ones in one place for now)
+// ===========================================================================
+constexpr Real lwc = 0.2302286341e-4;
+constexpr int lptr_so4_cw_amode[4] = {15, 23, 30, -1};
+constexpr int loffset = 9;
+constexpr Real small_value_lwc = 1.0e-8;
+constexpr Real small_value_cf = 1.0e-5;
+constexpr Real p0 = 101300.0;
+constexpr bool cloud_borne = false;
+constexpr bool modal_aerosols = false;
+// universal gas constant (sort of)
+// FIXME: TERRIBLE CONSTANT
+constexpr Real Ra = 8314.0 / 101325.0;
+// water acidity
+constexpr Real xkw = 1.0e-14;
+// FIXME: AWFUL CONSTANT
+// [cm3/L / Avogadro constant]
+constexpr Real const0 = 1.0e3 / 6.023e23;
+// 330 ppm = 330.0e-6 atm
+constexpr Real co2g = 330.0e-6;
+constexpr int itermax = 20;
+constexpr Real small_value_20 = 1.0e-20;
+constexpr Real small_value_30 = 1.0e-30;
+constexpr int numptrcw_amode[nmodes] = {23, 28, 36, 40};
+// ===========================================================================
+
+// struct for the sox_cldaero_create_obj()
+// NOTE: maybe ditch this in the future to make life simpler?
```

Comment:
Is this struct based on a similar Fortran derived type?

Reply:
It is. [In the fortran](https://github.com/eagles-project/e3sm_mam4_refactor/blob/6bbb278a86b047e4804e14eab705109dd9f98c00/components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90#L91-L126), it's a pointer to a derived type... or at least something like that.

I wasn't sure whether the struct makes it more or less readable, in favor of staying same-ish in comparison to MAM4.

Do you have strong feelings on this struct vs. individual variables?

Reply:
I think this is in line with our charge to port MAM4 directly. Fortran derived types and C-style structs are functionally identical. Personally, I hesitate even to opine on these kinds of details for MAM4 because I think its structure is dominated by historical accretion and it deserves a complete rethink.